Improvment on how to build from source (#3217)
mamba-org/mamba: docs/source/developer_zone/dev_environment.rst

`LSP <https://microsoft.github.io/language-server-protocol/>`_ code completion and lint.

> Build using CMake
> ===============================

Reviewer: They need to be the same length in RST.
```suggestion
Build using CMake
===============
```

.. note::

   Even if not using Taskfile, the file ``Taskfile.dist.yml`` can provide insights on
   useful commands.
   The CI files in ``.github/workflow`` provide an alternative way of developing Mamba.

.. warning::

>    It is supposed to build mamba package from a freshly downloaded source code, building from source that has been previously built with a different cmake configuration will lead to inconsistent result for the current cmake configuration

Reviewer: The "stateful" aspect of CMake you noticed is part of how CMake work.
https://cliutils.gitlab.io/modern-cmake/chapters/basics/variables.html

>    It is supposed to build mamba package from a freshly downloaded source code, building from source that has been previously built with a different cmake configuration will lead to inconsistent result for the current cmake configuration

Reviewer: I'm in favor of removing that warning completely since this is normal behavior for CMake.

Install develoment dependencies
*******************************

First create a virtual environment with necessary dependencies using  ``micromamba`` like follows:
.. code:: bash

    micromamba create -n mamba -c conda-forge -f dev/environment-dev.yml

Then activate the created a virtual environment with necessary dependencies using  ``micromamba`` like follows:
.. code:: bash

>     micromamba activate -n mamba

Reviewer: I am actually in favor a explicit `-n` here.

Compile ``libmmaba`` and other artifacts
****************************************

``libmamba`` is built using CMake.
Typically during development, we build everything dynamically using dependencies from Conda-Forge.

The first step is to configure the build options.

.. note::
    All ``cmake`` commands listed below use ``bash`` multi-line syntax.
    On Windows, replace ``\`` trailing character with ``^``.

A recommended set is already provided as CMake Preset, so feel free to use any variations.

.. code:: bash

    cmake -B build/ -G Ninja --preset mamba-unix-shared-debug-dev

> Compilation can then be launched as follows:

Reviewer 2: I prefer the original wording.

``libmamba`` tests
******************

> The tests for libamba are written in C++ and their corresponding executables should be located at the following path once the compilation finishes:

Reviewer 2: ```suggestion
The tests for libmamba are written in C++ and the corresponding executable to be run should be located at the following path once the compilation finishes:
```

.. code:: bash

    ./build/libmamba/tests/test_libmamba

``micromamba`` integration tests
********************************

Many ``micromamba`` integration tests are written through a Pytest Python wrapper.
The environment variable ``TEST_MAMBA_EXE`` controls which executable is being tested.

> First check out the resulting executable after the copilationa like follows:

Reviewer 2: ```suggestion
First check out the generated executable after the compilation is done:
```

> First check out the resulting executable after the copilationa like follows:
> 
> .. code:: bash
> 
>   $ ls build/micromamba/
>   CMakeFiles  cmake_install.cmake  mamba
> 

Reviewer: ```suggestion
```

I am not in favor of adding this paragraph. The files should be there under the given configuration and the micromamba test will emit an error if `TEST_MAMBA_EXE` points to something that does not exsist.

Author: Maybe not, since people could have already isntalled micromamba before building, and in the previous version it is said to export `TEST_MAMBA_EXE=micromamba`, I added this just to let people to be aware of any possible name change later.

Reviewer: I'm not sure to which version you refer, could you provide the link?
Otherwise, similarly to the C++ test section, we could just say somthing like: "The compiled executable get placed under `build/micromamba/micromamba" and moving on "we point the environment variable ``TEST_MAMBA_EXE`` to this location to set the executable under test" and move on to the next code section.

Author: At this link: https://mamba.readthedocs.io/en/latest/developer_zone/dev_environment.html, under the ` micromamba integration tests` section.
Changing to "we point the environment variable TEST_MAMBA_EXE to this location to set the executable under test"  seems to be much better for me.

Author: ```suggestion
The compiled executable will be placed at `build/micromamba/` so we point the environment variable `TEST_MAMBA_EXE` to this location to set the executable under test

```

> Then set the environment variable ``TEST_MAMBA_EXE`` as follows:

Reviewer 2: ```suggestion
Then set the environment variable ``TEST_MAMBA_EXE``:
```

.. code:: bash

   export TEST_MAMBA_EXE="${PWD}/build/micromamba/mamba"

> We can now run all ``micromamba`` integration tests like following:

Reviewer 2: ```suggestion
We can now run all ``micromamba`` integration tests:
```

   This needs to be done every time ``libamamba`` changes.

> First build python bindings like follows:

Reviewer 2: ```suggestion
First build python bindings:
```

.. code:: bash

    cmake --install build/ --prefix "${CONDA_PREFIX}"

> Then the Python bindings can be installed as follows:

Reviewer 2: You can keep the original wording :)

> Then the Python bindings can be installed as follows:

Reviewer 2: ```suggestion
Then the Python bindings can be installed:
```

.. code:: bash

    python -m pip install --no-deps --no-build-isolation libmambapy/

> Finally the tests can be invoked as follows:

Reviewer 2: ```suggestion
Finally the tests can be run:
```
